### PR TITLE
feat: upgraded to .net v8

### PR DESCRIPTION
**Pull request by sekmenhuseyin**
**What this PR does / why it's submitted / why we need it**:

---

**Checklist**:

- [ ] My PR has a descriptive title (not a vague title like `Needed this`)
- [ ] Title and all the commits follow [commitlint guidelines](https://github.com/conventional-changelog/commitlint#what-is-commitlint)
- [ ] My PR targets the `develop` branch or appropriate `release` branch
- [ ] My PR provides test coverage for my change (where applicable)
- [ ] I've placed the link to the corresponding Trello card or issue which this PR fixes/address below ⤵️

---

Fixes: ### ISSUE_OR_TRELLO_LINK ###

### CODE REVIEW

File: src/Cogworks.AzureSearch.IoC.Umbraco/Cogworks.AzureSearch.IoC.Umbraco.csproj [hunk 1]
```diff
@@ -15,18 +15,18 @@
         <None Include="..\..\README.md" Pack="true" PackagePath="\" />
     </ItemGroup>
 
+    <ItemGroup Condition="'$(TargetFramework)' == 'net8.0'">
+        <PackageReference Include="Umbraco.Cms.Core" Version="[12.0.0,)" />
```
Comment (aochmann) [inline src/Cogworks.AzureSearch.IoC.Umbraco/Cogworks.AzureSearch.IoC.Umbraco.csproj]: 13+ should be

File: src/Cogworks.AzureSearch.IoC.Umbraco/Cogworks.AzureSearch.IoC.Umbraco.csproj [hunk 2]
```diff
@@ -15,18 +15,18 @@
         <None Include="..\..\README.md" Pack="true" PackagePath="\" />
     </ItemGroup>
 
+    <ItemGroup Condition="'$(TargetFramework)' == 'net8.0'">
+        <PackageReference Include="Umbraco.Cms.Core" Version="[13.3.2,)" />
```
Comment (aochmann) [inline src/Cogworks.AzureSearch.IoC.Umbraco/Cogworks.AzureSearch.IoC.Umbraco.csproj]: Please make from 13.0.0 as begging version

Comment (sekmenhuseyin) [reply src/Cogworks.AzureSearch.IoC.Umbraco/Cogworks.AzureSearch.IoC.Umbraco.csproj]: I tried actually but pipelines gave an error: https://github.com/thecogworks/Cogworks.AzureSearch.IoC.Umbraco/actions/runs/9417358219/job/25942374104
image

Comment (sekmenhuseyin) [reply src/Cogworks.AzureSearch.IoC.Umbraco/Cogworks.AzureSearch.IoC.Umbraco.csproj]: https://github.com/advisories/GHSA-74p6-39f2-23v3

Comment (aochmann) [reply src/Cogworks.AzureSearch.IoC.Umbraco/Cogworks.AzureSearch.IoC.Umbraco.csproj]: You can make the `NU1902` as a warning.

We can still say minimal Umbraco version and it's up to the project configuration to choose the current Umbraco version 🤔

**Pull request merged**